addons/website/static/src/client_actions/configurator/configurator.js

```diff
@@ -451,13 +417,35 @@ class ThemeSelectionScreen extends ApplyConfiguratorScreen {
     setup() {
         super.setup();
 
+        this.uiService = useService('ui');
         this.orm = useService('orm');
         this.state = useStore();
         this.themeSVGPreviews = [useRef('ThemePreview1'), useRef('ThemePreview2'), useRef('ThemePreview3')];
+        const proms = [];
 
-        onMounted(() => {
+        onMounted(async () => {
+            // Add a loading effect during the loading of the images inside the
+            // svgs.
+            this.uiService.block();
             this.state.themes.forEach((theme, idx) => {
-                $(this.themeSVGPreviews[idx].el).append(theme.svg);
+                // Transform the text svg into a svg element.
+                const svgEl = new DOMParser().parseFromString(theme.svg, 'image/svg+xml').documentElement;
+                for (const imgEl of svgEl.querySelectorAll('image')) {
+                    proms.push(new Promise((resolve, reject) => {
+                        imgEl.addEventListener('load', () => {
+                            resolve(imgEl);
+                        }, {once: true});
+                        imgEl.addEventListener('error', () => {
+                            reject(imgEl);
+                        }, {once: true});
+                    }));
+                }
+                $(this.themeSVGPreviews[idx].el).append(svgEl);
+            });
+            // When all the images inside the svgs are loaded then remove the
+            // loading effect.
+            Promise.all(proms).then(() => {
+                this.uiService.unblock();
```

Comment:
This is problematic, it will never be executed if one of the promise is rejected.
That's what I already faced in local, for some reason I got a key error on a theme image (probably wrong theme version) which made a single image out of all of the images not load (as it was not found / 404).
The loading would never stop and spin forever as no code would release it.

Need to replace it by `allSettled` I guess?